lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1644,7 +1642,25 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
       unsigned FwdDeclLine = 0;
       assert(SizeInBits ==
              CI.getTargetInfo().getPointerWidth(clang::LangAS::Default));
-      return createPointerSizedStruct(Scope, Decl->getNameStr(), L.File,
+      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes) {
+        auto *DIType = createStructType(
+            DbgTy, Decl, ClassTy, Scope, File, L.Line, SizeInBits, AlignInBits,
+            Flags, nullptr, llvm::dwarf::DW_LANG_Swift, MangledName);
+        auto SuperClassTy = ClassTy->getSuperclass();
+        if (DIType && SuperClassTy) {
+          auto SuperClassDbgTy = DebugTypeInfo::getFromTypeInfo(
+              SuperClassTy, IGM.getTypeInfoForUnlowered(SuperClassTy), IGM,
+              false);
+
+          llvm::DIType *SuperClassDITy = getOrCreateType(SuperClassDbgTy);
+          if (SuperClassDITy) {
```

Comment:
What does it mean for this to fail?

Reply:
`getOrCreateType` is quite complex so I don't know if succeeds 100% of the time, this is just to guarantee we don't dereference a null ptr.  I'll add an assert to document the fact that we don't expect this to fail (I'll do the same for the one you commented above as well).

Reply:
The LLVM coding style has a really weird aversion to using references. I think it would be fine to assert and not check here.